osu.Game/Screens/Edit/EditorTableBackground.cs

```diff
@@ -0,0 +1,325 @@
+
+        public int RowCount
+        {
+            get => rowCount;
+            set
+            {
+                rowCount = value;
+                Height = rowCount * ROW_HEIGHT;
+                Deselect();
+            }
+        }
+
+        private readonly Bindable<int> hoveredIndexBindable = new Bindable<int>(-1);
+
+        private readonly HoverSampleSet sampleSet;
+        private Bindable<double?> lastHoverPlaybackTime = null!;
+
+        public EditorTableBackground(HoverSampleSet sampleSet = HoverSampleSet.Default)
+        {
+            this.sampleSet = sampleSet;
+        }
+
+        private Sample? sampleHover;
+        private Sample? sampleClick;
+
+        [BackgroundDependencyLoader]
+        private void load(AudioManager audio, SessionStatics statics)
+        {
+            // see HoverSampleDebounceComponent
+            lastHoverPlaybackTime = statics.GetBindable<double?>(Static.LastHoverSoundPlaybackTime);
+
+            // see HoverSounds
+            sampleHover = audio.Samples.Get($@"UI/{sampleSet.GetDescription()}-hover")
+                          ?? audio.Samples.Get($@"UI/{HoverSampleSet.Default.GetDescription()}-hover");
+
+            // see HoverClickSounds
+            sampleClick = audio.Samples.Get($@"UI/{sampleSet.GetDescription()}-select")
+                          ?? audio.Samples.Get($@"UI/{HoverSampleSet.Default.GetDescription()}-select");
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            hoveredIndexBindable.BindValueChanged(i => onHoveredChanged(i.NewValue));
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (IsHovered)
+                hoveredIndexBindable.Value = getItemIndexAtMousePosition();
+        }
+
+        protected override bool OnHover(HoverEvent e) => true;
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            base.OnHoverLost(e);
+            hoveredIndexBindable.Value = -1;
+        }
+
+        private void onHoveredChanged(int hoveredIndex)
+        {
+            EditorTableBackgroundRow? newHovered = null;
+
+            foreach (var child in this)
+            {
+                if (child.Index == hoveredIndex)
+                {
+                    newHovered = child;
+                }
+                else
+                {
+                    if (child.State == RowState.Hovered)
+                        child.State = RowState.None;
+                }
+            }
+
+            if (hoveredIndex == -1)
+                return;
+
+            if (newHovered != null)
+            {
+                if (newHovered.State == RowState.None)
+                    newHovered.State = RowState.Hovered;
+            }
+            else
+                Add(new EditorTableBackgroundRow(hoveredIndex));
+
+            playHoverSample();
+        }
+
+        public EditorTableBackgroundRow Select(int itemIndex)
```

Comment:
I am not having the "row background" of a table having a separate selection method from the actual row. It's the row. The user is performing the operation of selecting the _row_. It is the _**row**_ you're selecting.